libc/resolve_path.c

```diff
@@ -475,6 +475,48 @@ TEST(resolve_path, symlink_rename)
 }
 
 
+/* check the behavior of the realpath function when it encounters pathname resolution of a symbolic link that result with a huge length */
+TEST(resolve_path, symlink_long_resolution)
+{
+	const char symNameShort[] = "symShort";
+	char symNameLong[NAME_MAX / 2];
+	char resolved[PATH_MAX + 1];
+	char basePath[PATH_MAX - 5] = "symShort/";
+	int basePathLen = strlen(basePath);
+	const char *pathSegment = "dev/../";
+	const int pathSegmentLen = sizeof(pathSegment);
+
+	while (basePathLen + pathSegmentLen < sizeof(basePath) - 5) {
+		strcat(basePath, pathSegment);
+		basePathLen += pathSegmentLen;
```

Comment:
**basePathLen is much bigger than an actual size of basePath. Note that you add sizeof each time, so 8 when in fact only 7 bytes are added.**